.github/workflows/enable-pack-build-and-test.yaml

```diff
@@ -0,0 +1,116 @@
+name: Enable Pack Build and Test
+
+on:
+  # Daily at 3 in the morning
+  schedule:
+    - cron: "0 3 * * *"
```

Comment:
OK. I created the secret. I'm almost ready to merge this.

@armab I think once a month is enough, assuming that the API call resets the 60 day timer. That assumption is also baked into this:
https://stackoverflow.com/a/68591566/1134951
https://github.com/invenia/KeepActionsAlive

So, can we try once a month, and then if that turns out not to work, we can tweak this?

Reply:
> What if we skipped this check and just blindly enable it every time this workflow runs? That way, the 60d clock would reset every time this runs.

I hope it'll work, but I'd doubt it's possible to re-enable not yet disabled GH workflow, considering the solution mentioned there and this code: https://github.com/invenia/KeepActionsAlive/blob/main/src/app.py#L73

Otherwise, we'd see similar solutions in the wild when the github repo could re-enable itself, instead of using the https://github.com/marketplace/actions/keepalive-workflow self-commit approach.

More than that, the CI may expire after 2 months in this repo too. With that, I think there's a need to use this workflow in the https://github.com/stackStorm-exchange/index, as we have automated commits in there and confidence it won't expire.

But we can try as is in this repo and see how it goes to verify our guesses.

Reply:
> > What if we skipped this check and just blindly enable it every time this workflow runs? That way, the 60d clock would reset every time this runs.
> 
> I hope it'll work, but I'd doubt it's possible to re-enable not yet disabled GH workflow, considering the solution mentioned there and the code: https://github.com/invenia/KeepActionsAlive/blob/main/src/app.py#L73
> 
> Otherwise, we'd see similar solutions in the wild when the github repo could re-enable itself, instead of using the https://github.com/marketplace/actions/keepalive-workflow self-commit approach.
> 
> More than that, the GH Actions may expire after 2 months in this repo too. With that, I think there's a need to use this workflow in the https://github.com/stackStorm-exchange/index, as we have automated commits in there and confidence it won't expire.
> 
> But we can try as is in this repo and see how it goes to verify our guesses.

Yeah, my suspicion is that this "self-enabling" approach won't work either; using the index isn't a terrible approach since the index and ci repositories are functionally coupled anyhow, right?

Reply:
The index repo would be more convenient, true. But, I want to avoid adding a PAT on the index repo.

Right now, all the packs get pulled into the index CI. If someone compromises one of our packs, then the most they can do is mess up the index, they can't use the index to get to any other exchange packs.

If we add a PAT with write access to repos and workflows, then that widens such an attack vector. Then, if someone is able to compromise a pack and the index, they could jump from there to all the packs.

This CI repo is used in workflows across the exchange. Anyone that gets access to this repo already has access to all the packs. But, gaining access to any of the packs does not give any attack vector that could compromise this repo. So, I would prefer to have the PAT only on this repo.

(Hmm, maybe this workflow should avoid cloning all the packs, and rely on the GitHub API to look up everything about the available workflows).

If the API call doesn't work on this repo (I agree, there's a good chance it won't), then we can do one of three things:
- add dummy commits (doing it only in this repo, while ugly, is better than on all of the pack repos)
- add a workflow in st2 ci that deals with this repo (and only this repo, not the rest of the exchange)
- add a Circle I workflow that keeps this repo active.

I kind of like the idea of using CircleCI here. The token we add to CircleCI could use a fine grained token so that it can only hit the workflow enable endpoint on this one repo.

Reply:
> If we add a PAT with write access to repos and workflows

I don't think this would need write on the *repos* though, just on the workflows.

Reply:
All the good options :+1: 

- fine-grained PAT
- CircleCI
- automated commits in this repo

I'm fine either way

Reply:
Cool, deferring to @cognifloyd on which way to go!

Reply:
I'm merging it as is. Let's see how things go and then make changes in follow-ups